var/spack/repos/builtin/packages/namd/package.py

```diff
@@ -294,6 +294,12 @@ def edit(self, spec, prefix):
             if "+single_node_gpu" in spec:
                 opts.extend(["--with-single-node-cuda"])
 
+        if "+rocm" in spec:
+            self._copy_arch_file("hip")
+            opts.append("--with-hip")
+            opts.append("--with-single-node-hip")
```

Comment:
This requires a conflict between `+rocm` and `~single_node_gpu`, doesn't it?

Reply:
No, this does not require a conflict between ``+rocm`` and ``~single_node_gpu`` in this case, because I'm forcing NAMD to compile with the ``--whith-single-node-hip`` option when rocm is enabled in the code with ``+rocm``.
````
==> Concretized namd@3.0%gcc@12.1+rocm amdgpu_target==gfx90a interface=tcl arch=linux-rhel8-zen3 ^charmpp@7.0.0~cuda+omp~papi+production~pthreads+shared+smp+syncft~tcp+tracing backend=multicore build-target='charm++' ^cray-mpich@8.1.28 ^tcl@8.5.19
 -   x6svada  namd@3.0%gcc@12.1~cuda+rocm~single_node_gpu amdgpu_target=gfx90a build_system=makefile fftw=3 interface=tcl arch=linux-rhel8-zen3
 .
 .
 .
 ==> namd: Executing phase: 'edit'
==> namd: Executing phase: 'build'
==> namd: Executing phase: 'install'
==> namd: Successfully installed namd-3.0-x6svadajr3mxuk33lqp6ejzhqvqbb67r

````
I think I will make a change to be consistent with the rest of the code by adding : 

````
if "+rocm" in spec:
            self._copy_arch_file("hip")
            opts.append("--with-hip")
            opts.extend(["--rocm-prefix", os.environ["ROCM_PATH"]])
            if "+single_node_gpu" in spec:
                opts.extend(["--with-single-node-hip"])
            
````
in the added rocm section

Reply:
>  because I'm forcing NAMD to compile with the --whith-single-node-hip option when rocm is enabled in the code with +rocm

I was suggesting a conflict precisely for this reason. Otherwise the `+rocm+single_node_gpu` and `+rocm~single_node_gpu` are both allowed and will result in the same installation, which is confusing and `+rocm~single_node_gpu` is essentially meaningless.

I like the suggestion of making this consistent with what it is done for the `+cuda` variant, if `+rocm~single_node_gpu` is actually supported.